draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1221,96 +1235,145 @@ these unknown fields.
 
 # Serializing qlog {#concrete-formats}
 
-This document and other related qlog schema definitions are intentionally
-independent of serialization format. This means that implementers themselves can
-choose how to represent and serialize qlog data practically on disk or on the
-wire. Some examples of possible formats are JSON, CBOR, CSV, protobuf,
-flatbuffers, etc.
-
-All these formats make certain tradeoffs between flexibility and efficiency, with
-textual formats like JSON typically being more flexible but also less efficient
-than binary formats like protocol buffers. The format choice will depend on the
-practical use case of the qlog user. For example, for use in day to day debugging,
-a plaintext readable (yet relatively large) format like JSON is probably
-preferred. However, for use in production, a more optimized yet restricted format
-can be better. In this latter case, it will be more difficult to achieve
-interoperability between qlog implementations of various protocol stacks, as some
-custom or tweaked events from one might not be compatible with the format of the
-other. This will also reflect in tooling: not all tools will support all formats.
-
-This being said, the authors prefer JSON as the basis for storing qlog,
-as it retains full flexibility and maximum interoperability. Storage
-overhead can be managed well in practice by employing compression. For
-this reason, this document details how to practically transform qlog
-schema definitions to {{!JSON=RFC8259}}, its subset {{!I-JSON=RFC7493}},
-and its streamable derivative {{!JSON-Text-Sequences=RFC7464}}s. Concrete options
-to bring down JSON size and processing overheads are discuseed in {{optimizations}}.
-
-As depending on the employed format different deserializers/parsers should be
-used, the "qlog_format" field is used to indicate the chosen serialization
-approach. This field is always a string, but can be made hierarchical by the use
-of the "." separator between entries. For example, a value of "JSON.optimizationA"
-can indicate that a default JSON format is being used, but that a certain
-optimization of type A was applied to the file as well (see also
-{{optimizations}}).
+qlog schema definitions in this document, {{QLOG-QUIC}} and {{QLOG-H3}} are
```

Comment:
Right but it's too wooly to say "other related schema" - all extensions of qlog are related to the main schema. So the question becomes are we levying recommendations on additional schema to be generic, or can they make optimization for serialization formats? The latter seems reasonable to me, it's part of tradeoff that we mention in this spec.

 We have two examples that are generic but don't think we can extapolate from that. This might lean towards having a serialization section in each of the documents so that they state if they are generic or specific. I created #357 to track that.